cmd/nerdctl/login.go

```diff
@@ -287,7 +289,14 @@ func tryLoginWithRegHost(ctx context.Context, rh docker.RegistryHost) error {
 		}
 		res, err := ctxhttp.Do(ctx, rh.Client, req)
 		if err != nil {
-			return fmt.Errorf("failed to call rh.Client.Do: %w", err)
+			// In some circumstance (e.g. people just use 80 port to support pure http), the error will contain message like "dial tcp <port>: connection refused"
+			if (!imgutil.IsErrHTTPResponseToHTTPSClient(err) || !imgutil.IsThePortHasBeenClosed(err)) && insecure {
+				logrus.WithError(err).Warnf("server %q does not seem to support HTTPS, falling back to plain HTTP", rh.Host)
```

Comment:
May need to remind users that they have --insecure-registry flags

Reply:
Sounds good!